.teamcity/Gradle_Check/model/BucketProvider.kt

```diff
@@ -127,6 +127,25 @@ class StatisticBasedGradleBuildBucketProvider(private val model: CIBuildModel, t
             MAX_PROJECT_NUMBER_IN_BUCKET
         )
     }
+
+    private fun determineSubProjectClassTimes(testCoverage: TestCoverage, buildProjectClassTimes: BuildProjectToSubprojectTestClassTimes): Map<String, List<TestClassTime>>? {
+        val testCoverageId = testCoverage.asId(model)
+        return buildProjectClassTimes[testCoverageId] ?: if (testCoverage.testType == TestType.soak) {
+            null
+        } else {
+            val testCoverages = model.stages.flatMap { it.functionalTests }
+            val foundTestCoverage = testCoverages.firstOrNull {
+                it.testType == TestType.platform &&
+                    it.os == testCoverage.os &&
+                    it.buildJvmVersion == testCoverage.buildJvmVersion
+            }
+            foundTestCoverage?.let {
+                buildProjectClassTimes[it.asId(model)]
+            }?.also {
+                println("No test statistics found for ${testCoverage.asName()} (${testCoverage.uuid}), re-using the data from ${foundTestCoverage.asName()} (${foundTestCoverage.uuid})")
```

Comment:
This message is only printed during `mvn teamcity:generate-configs`, I'm wondering if we can see this message anywhere on TC.

Reply:
Yeah, I also don't know where to see the messages on CI. Seeing them for `mvn teamcity:generate-configs` was helpful, though.